File: SpringBootJpa/src/test/java/com/example/java/EntityManagerTests.java
package com.example.java;

import com.example.java.simple.domain.Customer;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;

import static org.junit.jupiter.api.Assertions.assertNotEquals;

@DataJpaTest
class EntityManagerTests {

    @Autowired
    private TestEntityManager testEntityManager;

    // 비영속성 데이터
    private Customer getPersistenceContextCustomer() {
        return new Customer("heo won chul", "010-xxxx-xxxx", "developer");
    }

    @Test
    void test_insertClearAndFindAndUpdateClear() {
        Customer customer = testEntityManager.persistFlushFind(getPersistenceContextCustomer());
        customer.changeBigo("Developer");
        testEntityManager.flush(); // Database 동기화
//		  testEntityManager.clear(); // Persistence Context 초기화

        Customer result = testEntityManager.find(Customer.class, customer.getIdx());
        assertNotEquals("developer", result.getBigo());
    }
}

File: SpringBootJpa/src/test/kotlin/com/example/kotlin/EntityManagerTests.kt
package com.example.kotlin

import com.example.kotlin.simple.domain.Customer
import org.junit.Assert.assertNotEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager

@DataJpaTest
class EntityManagerTests {

    @Autowired
    lateinit var testEntityManager: TestEntityManager

    @Test
    fun test_insertClearAndFindAndUpdateClear() {
        val customer = testEntityManager.persistFlushFind(com.example.kotlin.simple.domain.Customer(name = "heo won chul", tel = "010-xxxx-xxxx", bigo = "developer")) // 비영속성 데이터
        customer.changeBigo("Developer")
        testEntityManager.flush() // Database 동기화
        //		  testEntityManager.clear(); // Persistence Context 초기화

        val result = testEntityManager.find(Customer::class.java, customer.idx)
        assertNotEquals(result.bigo, "developer")
    }
}
